apps/mark-scan/accessible-controller/src/controllerd.rs

```diff
@@ -303,42 +336,109 @@ fn create_virtual_device() -> Device {
 }
 
 fn main() {
-    // Open the serial port
-    let port = serialport::new(DEVICE_PATH, DEVICE_BAUD_RATE)
-        .timeout(POLL_INTERVAL)
-        .open();
-
-    println!("Opened controller serial port at {DEVICE_PATH}");
+    log(Log {
+        event_id: "process-started".to_string(),
+        message: format!("Opened controller serial port at {DEVICE_PATH}"),
+        event_type: EventType::SystemAction,
+        ..Default::default()
+    });
+
+    let running = Arc::new(AtomicBool::new(true));
+    let r = running.clone();
+
+    if let Err(e) = ctrlc::set_handler(move || {
+        r.store(false, Ordering::SeqCst);
```

Comment:
I tend to prefer this style where I have a closure that needs a clone of an outer binding:

```suggestion
    if let Err(e) = ctrlc::set_handler({
        let running = running.clone();
        move || {
            running.store(false, Ordering::SeqCst);
        }
```
That way you keep the same name and you can remove the `r` from the outer scope that was only ever needed in this closure.